src/pages/iou/IOUModal.js

```diff
@@ -82,6 +91,42 @@ class IOUModal extends Component {
         }));
     }
 
+    /**
+     * Update amount with number or Backspace pressed.
+     * Validate new amount with decimal number regex up to 6 digits and 2 decimal digit
+     *
+     * @param {String} buttonPressed
+     */
+    updateAmount(buttonPressed) {
+        // Backspace button is pressed
+        if (buttonPressed === '<' || buttonPressed === 'Backspace') {
+            if (this.state.amount.length > 0) {
+                this.setState(prevState => ({
+                    amount: prevState.amount.substring(0, prevState.amount.length - 1),
+                    isAmountPageNextButtonDisabled: prevState.amount.length === 1,
+                }));
+            }
+        } else {
+            const decimalNumberRegex = new RegExp(/^\d{1,6}(\.\d{0,2})?$/, 'i');
```

Comment:
Could we move the regex pattern to `CONST.js` please, as this will be useful elsewhere in the future.

Reply:
Actually, please ignore this -- no need to do this until we actually start reusing it